lib/OpenQA/Worker/Cache.pm

```diff
@@ -129,6 +129,16 @@ sub download_asset {
             $asset = undef;
         }
     }
+    elsif ($tx->res->code && $tx->res->code =~ /5[0-9]{2}/) {
```

Comment:
is_server_error did the trick, however no error code is provided for situations where the the webserver is down. I used [531](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#Cloudflare) for that :)